### PR TITLE
Throw an error if an application tries to add streams, datasets,

### PR DESCRIPTION
**Pull request by gokulavasan**
plugins with same name in same/different programs.

Punting the more detailed error message to next release (JIRA : https://issues.cask.co/browse/CDAP-3729)

Build : http://builds.cask.co/browse/CDAP-RBT477

### CODE REVIEW

**Comment by gokulavasan** [comment]
@albertshau Please review

File: cdap-api/src/main/java/co/cask/cdap/internal/api/DefaultDatasetConfigurer.java [hunk 1]
```diff
@@ -50,21 +52,43 @@
...
   public void addDatasetSpecs(Map<String, DatasetCreationSpec> datasetSpecs) {
+    Set<String> duplicateDatasetSpecs = findDuplicates(datasetSpecs.keySet(), this.datasetSpecs.keySet());
+    checkArgument(duplicateDatasetSpecs.isEmpty(),
+                  "Dataset Instances %s have already been added. Remove the duplicates.", duplicateDatasetSpecs);
     this.datasetSpecs.putAll(datasetSpecs);
   }
 
   public void addDatasetModules(Map<String, String> datasetModules) {
+    for (Map.Entry<String, String> newEntry : datasetModules.entrySet()) {
+      if (datasetModules.containsKey(newEntry.getKey())) {
```
Comment (sagarkapare) [inline cdap-api/src/main/java/co/cask/cdap/internal/api/DefaultDatasetConfigurer.java]: should it be `this.datasetModules`?

Comment (gokulavasan) [inline cdap-api/src/main/java/co/cask/cdap/internal/api/DefaultDatasetConfigurer.java]: Yes, good catch! Thanks! Updated the PR and also enhanced the unit test to cover this case.

**Comment by gokulavasan** [comment]
@sagarkapare Please take a look again. Thanks!

**Comment by sagarkapare** [comment]
LGTM :+1:

**Comment by gokulavasan** [comment]
Build is :green_apple:. Merging.

**Pull request merged**